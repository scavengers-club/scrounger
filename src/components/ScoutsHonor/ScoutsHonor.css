.scout-button {
  width: 25vw;
  background-color: var(--purple);
  border-radius: 7px;
  padding: 3px;
  display: block;
  margin: 2vh;
}

.scout-text {
  display: flex;
  flex-direction: column;
  margin: 3vh;
}

.scout-text h3,
p {
  margin: 1vh;
}

.scout-text h3 {
  text-align: center;
  font-family: 'Reenie Beanie', cursive;
  font-size: 200%;
  text-transform: uppercase;
}

.lets-explore {
  display: flex;
  flex-direction: column;
  text-align: center;
  align-items: center;
  margin-top: 2vh;
}
