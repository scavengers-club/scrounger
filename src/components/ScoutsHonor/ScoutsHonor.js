import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';

//this page will appear only when a new user is created
//will prompt user to agree to basic terms and conitions, verify age, etc

export default function ScoutsHonor() {
  const [checked, setChecked] = useState(false);
  const history = useHistory();

  const handleSubmit = async () => {
    if (checked) {
      history.push('/');
    } else {
      alert('you must agree to scouts honor');
    }
  };

  return (
    <div>
      <p>
        Please be respectful of others and property. Leave things as you find them. Be responsible
        if using this app after dark. Carry a mask so as to keep yourself and others safe. If you
        are adventuring alone, make sure someone knows where you are. Happy Adventuring!
      </p>
      <label>
        <input type="checkbox" isChecked={checked} onChange={() => setChecked(!checked)} />
        <span>I have read and agree to the Scouts Honor Pledge</span>
      </label>
      <button onClick={handleSubmit}>Lets Explore</button>
    </div>
  );
}
