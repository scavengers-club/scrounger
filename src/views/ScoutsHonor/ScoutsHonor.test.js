import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import ScoutsHonor from './ScoutsHonor';

test('<ScoutsHonor/> should render the Pledge', () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/scoutshonor']}>
      <Route path="/scoutshonor">
        <ScoutsHonor />
      </Route>
    </MemoryRouter>
  );

  screen.getByText(
    `Please be respectful of others and property. Leave things as you find them. Be responsible if using this app after dark. Carry a mask so as to keep yourself and others safe. If you are adventuring alone, make sure someone knows where you are. Be sure to keep your phone charged.`
  );
  expect(container).toMatchSnapshot();
});
