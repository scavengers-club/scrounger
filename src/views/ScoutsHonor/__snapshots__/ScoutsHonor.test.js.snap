// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScoutsHonor/> should render the Pledge 1`] = `
<div>
  <div
    class="scout-text"
  >
    <h3>
      Getting Started
    </h3>
    <p>
      Once you have agreed to the Scout's Honor click on the button at the bottom of this page to begin.
    </p>
    <p>
      The homepage will reveal all of the adventures available to you - click on an adventure to view its details. Follow the hints to discover new sights and places. Want to create a new adventure? Head to your profile by selecting your avatar and get to creating!
    </p>
    <h3>
      Scout's Honor Pledge
    </h3>
    <p>
      Please be respectful of others and property. Leave things as you find them. Be responsible if using this app after dark. Carry a mask so as to keep yourself and others safe. If you are adventuring alone, make sure someone knows where you are. Be sure to keep your phone charged.
    </p>
    <p>
      Happy Adventuring!
    </p>
    <div
      class="lets-explore"
    >
      <label>
        <input
          type="checkbox"
          value="false"
        />
        <span>
          I have read and agree to the Scouts Honor Pledge
        </span>
      </label>
      <button
        class="scout-button"
      >
        Lets Explore
      </button>
    </div>
  </div>
</div>
`;
